OPHD/UI/Reports/MineReport.cpp

```diff
@@ -332,7 +335,81 @@ void MineReport::drawTruckMangementPane(const NAS2D::Point<int>& origin)
+	auto& routeTable = NAS2D::Utility<std::map<class MineFacility*, Route>>::get();
+	bool routeAvailable = routeTable.find(mFacility) != routeTable.end();
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ 0, 65 },
+		labelWidth,
+		"Route Available",
+		routeAvailable ? "Yes" : "No",
+		routeAvailable ? textColor : NAS2D::Color::Red);
+
+	if (routeAvailable)
+	{
+		drawTruckHaulInfo(origin + NAS2D::Vector{ 0, 80 });
+	}
+}
+
+
+void MineReport::drawTruckHaulInfo(const NAS2D::Point<int>& origin)
+{
+	auto& r = Utility<Renderer>::get();
+	const auto textColor = NAS2D::Color{ 0, 185, 0 };
+	auto& routeTable = NAS2D::Utility<std::map<class MineFacility*, Route>>::get();
+	const auto mFacility = static_cast<MineFacility*>(selectedFacility);
+
+	auto& route = routeTable[mFacility];
+	drawLabelAndValueRightJustify(origin,
+		btnAddTruck.positionX() - origin.x - 10,
+		"Route Cost",
+		std::to_string(route.cost).substr(0, std::to_string(route.cost).find(".") + 3), // hack-ish and probably slow, this could be cached
+		textColor);
+
+
+	const float routeCost = std::clamp(route.cost, 1.0f, FLT_MAX);
+	const int totalOreMovement = static_cast<int>(constants::ShortestPathTraversalCount / routeCost) * mFacility->assignedTrucks();
+	const int oreMovementLabelWidth = r.size().x - origin.x - 10;
+	const int oreMovementPart = totalOreMovement / 4;
+	const int oreLabelWidth = (oreMovementLabelWidth - 10) / 2;
+
+	const NAS2D::Rectangle<int> tableRect({ origin.x - 2, origin.y + 18, oreMovementLabelWidth + 5, 47 });
+
+	r.drawBoxFilled(tableRect, { 0, 0, 0, 100 });
+	r.drawBox(tableRect, textColor);
+
+	r.drawLine(origin + NAS2D::Vector{ 0, 34 }, origin + NAS2D::Vector{ oreMovementLabelWidth, 34 }, textColor);
+	r.drawLine(origin + NAS2D::Vector{ 0, 50 }, origin + NAS2D::Vector{ oreMovementLabelWidth, 50 }, textColor);
+	r.drawLine(origin + NAS2D::Vector{ oreLabelWidth + 5, 37 }, origin + NAS2D::Vector{ oreLabelWidth + 5, 63 }, textColor);
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ 0, 20 },
+		oreMovementLabelWidth,
+		"Total Haul Capacity per Turn",
+		std::to_string(totalOreMovement),
+		textColor);
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ 0, 35 },
+		oreLabelWidth,
+		ResourceNamesOre[0] + " Haul Capacity",
+		std::to_string(oreMovementPart),
+		textColor);
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ oreLabelWidth + 10, 35 },
+		oreLabelWidth,
+		ResourceNamesOre[1] + " Haul Capacity",
+		std::to_string(oreMovementPart),
+		textColor);
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ 0, 50 },
+		oreLabelWidth,
+		ResourceNamesOre[2] + " Haul Capacity",
+		std::to_string(oreMovementPart),
+		textColor);
+
+	drawLabelAndValueRightJustify(origin + NAS2D::Vector{ oreLabelWidth + 10, 50 },
+		oreLabelWidth,
+		ResourceNamesOre[3] + " Haul Capacity",
+		std::to_string(oreMovementPart + (totalOreMovement % 4)),
+		textColor);
```

Comment:
I'm not too crazy about the indentation here either but this is what Visual Studio suggested as I was breaking it up and I stuck with it. I could have left it all on one line but it just becomes difficult to read at that point.

Anyway, I figured since the editor did the indentation here I'd not fiddle around with it to keep it consistent.

Reply:
Makes sense, I've seen a lot of Microsoft code indented this way. Later on I've seen other styles which I've found easier to read.

I would recommend spilling all parameters, including the first. Indentation should be a single level increase. I might also recommend putting the closing parenthesis on it's own line, much like is done with braces.

Example:
```cpp
drawLabelAndValueRightJustify(
	origin + NAS2D::Vector{ oreLabelWidth + 10, 50 },
	oreLabelWidth,
	ResourceNamesOre[3] + " Haul Capacity",
	std::to_string(oreMovementPart + (totalOreMovement % 4)),
	textColor
);
```